### PR TITLE
#349: onLogic enpoint

### PR DESCRIPTION
**Pull request by BeeMargarida**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-white/issues/349 |
| Dependencies | -- |
| Decisions | Call methods from the logic file of a build |
| Animated GIF | |

### CODE REVIEW

File: src/js/api/brand.js [hunk 1]
```diff
@@ -392,6 +430,34 @@ ripe.Ripe.prototype._getFactoryOptions = function(options = {}) {
     });
 };
 
+/**
+ * @ignore
+ * @see {link http://docs.platforme.com/#product-endpoints-factory}
```
Comment (gcandal) [inline src/js/api/brand.js:435]: oops

File: src/js/api/brand.js [hunk 2]
```diff
@@ -164,6 +164,44 @@ ripe.Ripe.prototype.getFactoryP = function(options, callback) {
...
+ * Returns the factory information where a model is made,
+ * specifically its name and the estimated production time in days.
+ * If no model is provided then returns the defaults of the owner's current model.
+ *
+ * @param {Object} options An object with options, such as:
+ *  - 'brand' - The brand of the model
+ *  - 'model' - The name of the model
+ * @param {Function} callback Function with the result of the request.
+ * @returns {XMLHttpRequest} The factory information for the provided model.
+ */
```
Comment (gcandal) [inline src/js/api/brand.js:177]: Update

File: src/js/api/brand.js [hunk 3]
```diff
@@ -164,6 +164,44 @@ ripe.Ripe.prototype.getFactoryP = function(options, callback) {
...
+ * Returns the factory information where a model is made,
+ * specifically its name and the estimated production time in days.
+ * If no model is provided then returns the defaults of the owner's current model.
+ *
+ * @param {Object} options An object with options, such as:
+ *  - 'brand' - The brand of the model
+ *  - 'model' - The name of the model
+ * @param {Function} callback Function with the result of the request.
+ * @returns {Promise} The factory information for the provided model.
+ */
```
Comment (gcandal) [inline src/js/api/brand.js:196]: Update

File: src/js/api/brand.js [hunk 4]
```diff
@@ -164,6 +164,42 @@ ripe.Ripe.prototype.getFactoryP = function(options, callback) {
     });
 };
 
+/**
+ * Returns the result of a given method of the logic script of a model.
+ *
+ * @param {Object} options An object with options, such as:
```
Comment (gcandal) [inline src/js/api/brand.js:170]: `version` missing

Comment (gcandal) [reply src/js/api/brand.js:170]: `data` missing

File: src/js/api/brand.js [hunk 5]
```diff
@@ -392,6 +428,23 @@ ripe.Ripe.prototype._getFactoryOptions = function(options = {}) {
     });
 };
 
+ripe.Ripe.prototype._getLogicOptions = function(options = {}) {
+    const brand = options.brand === undefined ? this.brand : options.brand;
+    const model = options.model === undefined ? this.model : options.model;
+    const version = options.version === undefined ? this.version : options.version;
+    const method = options.method === undefined ? null : options.method;
+    const url = this.url + "brands/" + brand + "/models/" + model + "/logic/" + method;
```
Comment (gcandal) [inline src/js/api/brand.js:436]: Use string interpolation with backticks instead, it makes it easier to read.

File: src/js/api/brand.js [hunk 6]
```diff
@@ -392,6 +428,23 @@ ripe.Ripe.prototype._getFactoryOptions = function(options = {}) {
...
 
+ripe.Ripe.prototype._getLogicOptions = function(options = {}) {
+    const brand = options.brand === undefined ? this.brand : options.brand;
+    const model = options.model === undefined ? this.model : options.model;
+    const version = options.version === undefined ? this.version : options.version;
+    const method = options.method === undefined ? null : options.method;
+    const url = this.url + "brands/" + brand + "/models/" + model + "/logic/" + method;
+    const params = {};
+    if (version !== undefined && version !== null) {
+        params.version = version;
```
Comment (gcandal) [inline src/js/api/brand.js:439]: Did you test this is working? Not saying it's not, but please double-check.

File: src/js/api/brand.js [hunk 7]
```diff
@@ -392,6 +428,23 @@ ripe.Ripe.prototype._getFactoryOptions = function(options = {}) {
...
+    }
+    return Object.assign(options, {
+        url: url,
+        method: "POST",
+        params: params
+    });
```
Comment (gcandal) [inline src/js/api/brand.js:445]: `data` missing.

File: src/js/api/brand.js [hunk 8]
```diff
@@ -392,6 +428,23 @@ ripe.Ripe.prototype._getFactoryOptions = function(options = {}) {
     });
 };
 
+ripe.Ripe.prototype._getLogicOptions = function(options = {}) {
```
Comment (gcandal) [inline src/js/api/brand.js:431]: Add jsdocs, even it's just an `@ignore`

**Comment by joamag** [comment]
@BeeMargarida @gcandal renmaed to `runLogic` and `runLogicP`

**Pull request merged**